src/components/GiveProject/GrantCard.tsx

```diff
@@ -610,7 +485,7 @@ export default function GrantCard({ grant, mode }: GrantDetailsProps) {
                             </Grid>
                             <Grid item>
                               <Typography className="metric">
-                                {donationInfo[donationId]
+                                {donationId != NO_DONATION && donationInfo[donationId]
```

Comment:
You could probably simplify this and use something like:

```ts
const currentDonation = useMemo(() => {
   if (donationId == NO_DONATION) return null;

   return donationInfo[donationId];
}, [donationInfo, donationId]);
```
(please excuse typos)

then this (and subsequent) conditionals change to:
```ts
currentDonation ? parseFloat(currentDonation.yieldDonated).toFixed(2) : "0"
```